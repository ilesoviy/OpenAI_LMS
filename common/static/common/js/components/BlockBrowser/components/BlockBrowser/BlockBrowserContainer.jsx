import { connect } from 'react-redux';
import { changeRoot } from '../../data/actions/courseBlocks';
import { getActiveBlockTree } from '../../data/selectors/index';
import { BlockBrowser } from './BlockBrowser';

const mapStateToProps = state => ({
    blocks: getActiveBlockTree(state),
    selectedBlock: state.selectedBlock,
});

const mapDispatchToProps = dispatch => ({
    onChangeRoot: blockId => dispatch(changeRoot(blockId)),
});

const BlockBrowserContainer = connect(
    mapStateToProps,
    mapDispatchToProps,
)(BlockBrowser);

export default BlockBrowserContainer;
