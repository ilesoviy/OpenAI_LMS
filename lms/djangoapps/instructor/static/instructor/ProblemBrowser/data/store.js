import { applyMiddleware, createStore } from 'redux';
import thunkMiddleware from 'redux-thunk';

import rootReducer from './reducers';

const configureStore = initialState => createStore(
    rootReducer,
    initialState,
    applyMiddleware(thunkMiddleware),
);

const store = configureStore();

export default store;
