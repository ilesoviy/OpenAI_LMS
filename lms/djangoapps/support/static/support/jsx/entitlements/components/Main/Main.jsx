import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Button, StatusAlert } from '@edx/paragon';
import SearchContainer from '../Search/SearchContainer';
import EntitlementSupportTableContainer from '../Table/EntitlementSupportTableContainer';
import EntitlementFormContainer from '../EntitlementForm/container';

const Main = props => (
    <div className="entitlement-support-wrapper">
        <StatusAlert
            alertType="danger"
            dialog={props.errorMessage}
            onClose={props.dismissErrorMessage}
            open={!!props.errorMessage}
        />
        <h2>
            Student Support: Entitlement
        </h2>
        <MainContent
            isFormOpen={props.isFormOpen}
            ecommerceUrl={props.ecommerceUrl}
            openCreationForm={props.openCreationForm}
            entitlements={props.entitlements}
        />
    </div>
);

const MainContent = (props) => {
    if (props.isFormOpen) {
        return <EntitlementFormContainer />;
    }

    return (
        <div>
            <div className="actions">
                <SearchContainer />
                <Button
                    className={['btn', 'btn-primary']}
                    label="Create New Entitlement"
                    onClick={props.openCreationForm}
                />
            </div>
            {
                props.entitlements.length > 0
                    ? <EntitlementSupportTableContainer ecommerceUrl={props.ecommerceUrl} /> : null
            }
        </div>
    );
};

const mapStateToProps = state => ({
    entitlements: state.entitlements,
});

Main.propTypes = {
    errorMessage: PropTypes.string.isRequired,
    dismissErrorMessage: PropTypes.func.isRequired,
    openCreationForm: PropTypes.func.isRequired,
    ecommerceUrl: PropTypes.string.isRequired,
    isFormOpen: PropTypes.bool.isRequired,
    entitlements: PropTypes.arrayOf(PropTypes.string).isRequired,
};

MainContent.propTypes = {
    openCreationForm: PropTypes.func.isRequired,
    ecommerceUrl: PropTypes.string.isRequired,
    isFormOpen: PropTypes.bool.isRequired,
    entitlements: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default connect(mapStateToProps, null)(Main);
