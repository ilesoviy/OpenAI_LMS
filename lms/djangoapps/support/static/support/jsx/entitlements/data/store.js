import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';

import rootReducer from './reducers/index';

const defaultState = {
    entitlements: [],
    error: '',
    form: {
        formType: '',
        isOpen: false,
        activeEntitlement: null,
    },
};

const configureStore = initialState => createStore(
    rootReducer,
    initialState,
    applyMiddleware(thunkMiddleware),
);

const store = configureStore(defaultState);

export default store;
