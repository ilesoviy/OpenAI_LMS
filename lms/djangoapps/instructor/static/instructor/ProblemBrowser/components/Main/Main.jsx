/* global gettext */
import { Button, Icon } from '@edx/paragon';
import BlockBrowserContainer from 'BlockBrowser/components/BlockBrowser/BlockBrowserContainer';
import * as PropTypes from 'prop-types';
import * as React from 'react';
import { ReportStatusContainer } from '../ReportStatus/ReportStatusContainer';

export default class Main extends React.Component {
    constructor(props) {
        super(props);
        this.handleToggleDropdown = this.handleToggleDropdown.bind(this);
        this.initiateReportGeneration = this.initiateReportGeneration.bind(this);
        this.state = {
            showDropdown: false,
        };
    }

    handleToggleDropdown() {
        this.props.fetchCourseBlocks(this.props.courseId, this.props.excludeBlockTypes);
        this.setState({ showDropdown: !this.state.showDropdown });
    }

    hideDropdown() {
        this.setState({ showDropdown: false });
    }

    initiateReportGeneration() {
        this.props.createProblemResponsesReportTask(
            this.props.problemResponsesEndpoint,
            this.props.taskStatusEndpoint,
            this.props.reportDownloadEndpoint,
            this.props.selectedBlock,
        );
    }

    render() {
        const { selectedBlock, onSelectBlock } = this.props;
        let selectorType = <Button onClick={this.handleToggleDropdown} label={gettext('Select a section or problem')} />;
        if (this.props.showBtnUi === 'false') {
            selectorType = (
            // eslint-disable-next-line jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events
                <span
                    onClick={this.handleToggleDropdown}
                    className={['problem-selector']}
                >
                    <span>{selectedBlock || 'Select a section or problem'}</span>
                    <span className={['pull-right']}>
                        <Icon
                            className={['fa', 'fa-sort']}
                        />
                    </span>
                </span>
            );
        }

        return (
            <div className="problem-browser-container">
                <div className="problem-browser">
                    {selectorType}
                    <input
                        type="text"
                        name="problem-location"
                        value={selectedBlock}
                        disabled
                        hidden={this.props.showBtnUi === 'false'}
                    />
                    {this.state.showDropdown
            && (
                <BlockBrowserContainer
                    onSelectBlock={(blockId) => {
                        this.hideDropdown();
                        onSelectBlock(blockId);
                    }}
                />
            )}
                    <Button
                        onClick={this.initiateReportGeneration}
                        name="list-problem-responses-csv"
                        label={gettext('Create a report of problem responses')}
                    />
                    <ReportStatusContainer />
                </div>

            </div>
        );
    }
}

Main.propTypes = {
    courseId: PropTypes.string.isRequired,
    createProblemResponsesReportTask: PropTypes.func.isRequired,
    excludeBlockTypes: PropTypes.arrayOf(PropTypes.string),
    fetchCourseBlocks: PropTypes.func.isRequired,
    problemResponsesEndpoint: PropTypes.string.isRequired,
    onSelectBlock: PropTypes.func.isRequired,
    selectedBlock: PropTypes.string,
    taskStatusEndpoint: PropTypes.string.isRequired,
    reportDownloadEndpoint: PropTypes.string.isRequired,
    ShowBtnUi: PropTypes.string.isRequired,
};

Main.defaultProps = {
    excludeBlockTypes: null,
    selectedBlock: '',
    timeout: null,
};
