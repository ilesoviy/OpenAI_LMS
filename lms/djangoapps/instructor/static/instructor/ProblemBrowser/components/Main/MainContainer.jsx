import { fetchCourseBlocks, selectBlock } from 'BlockBrowser/data/actions/courseBlocks';
import { connect } from 'react-redux';
import { createProblemResponsesReportTask } from '../../data/actions/problemResponses';
import Main from './Main';

const mapStateToProps = state => ({
    selectedBlock: state.selectedBlock,
});

const mapDispatchToProps = dispatch => ({
    onSelectBlock: blockId => dispatch(selectBlock(blockId)),
    fetchCourseBlocks:
    (courseId, excludeBlockTypes) => dispatch(fetchCourseBlocks(courseId, excludeBlockTypes)),
    createProblemResponsesReportTask:
    (problemResponsesEndpoint, taskStatusEndpoint, reportDownloadEndpoint, problemLocation) => dispatch(
        createProblemResponsesReportTask(problemResponsesEndpoint, taskStatusEndpoint, reportDownloadEndpoint, problemLocation),
    ),
});

const MainContainer = connect(
    mapStateToProps,
    mapDispatchToProps,
)(Main);

export default MainContainer;
