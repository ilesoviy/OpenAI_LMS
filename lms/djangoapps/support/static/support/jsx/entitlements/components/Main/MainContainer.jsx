import { connect } from 'react-redux';

import { dismissError } from '../../data/actions/error';
import { openCreationForm } from '../../data/actions/form';

import Main from './Main.jsx';

const mapStateToProps = state => ({
    errorMessage: state.error,
    isFormOpen: state.form.isOpen,
});

const mapDispatchToProps = dispatch => ({
    dismissErrorMessage: () => dispatch(dismissError()),
    openCreationForm: () => dispatch(openCreationForm()),
});

const MainContainer = connect(
    mapStateToProps,
    mapDispatchToProps,
)(Main);

export default MainContainer;
