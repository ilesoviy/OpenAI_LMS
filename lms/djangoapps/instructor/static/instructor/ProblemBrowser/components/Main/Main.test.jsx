/* global jest,test,describe,expect */
import { Button } from '@edx/paragon';
import BlockBrowserContainer from 'BlockBrowser/components/BlockBrowser/BlockBrowserContainer';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import store from '../../data/store';

import Main from './Main';

describe('ProblemBrowser Main component', () => {
    const courseId = 'testcourse';
    const problemResponsesEndpoint = '/api/problem_responses/';
    const taskStatusEndpoint = '/api/task_status/';
    const excludedBlockTypes = [];

    test('render with basic parameters', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Main
                    courseId={courseId}
                    createProblemResponsesReportTask={jest.fn()}
                    excludeBlockTypes={excludedBlockTypes}
                    fetchCourseBlocks={jest.fn()}
                    problemResponsesEndpoint={problemResponsesEndpoint}
                    onSelectBlock={jest.fn()}
                    selectedBlock={null}
                    taskStatusEndpoint={taskStatusEndpoint}
                />
            </Provider>,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render with selected block', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Main
                    courseId={courseId}
                    createProblemResponsesReportTask={jest.fn()}
                    excludeBlockTypes={excludedBlockTypes}
                    fetchCourseBlocks={jest.fn()}
                    problemResponsesEndpoint={problemResponsesEndpoint}
                    onSelectBlock={jest.fn()}
                    selectedBlock="some-selected-block"
                    taskStatusEndpoint={taskStatusEndpoint}
                />
            </Provider>,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('fetch course block on toggling dropdown', () => {
        const fetchCourseBlocksMock = jest.fn();
        const component = renderer.create(
            <Provider store={store}>
                <Main
                    courseId={courseId}
                    createProblemResponsesReportTask={jest.fn()}
                    excludeBlockTypes={excludedBlockTypes}
                    fetchCourseBlocks={fetchCourseBlocksMock}
                    problemResponsesEndpoint={problemResponsesEndpoint}
                    onSelectBlock={jest.fn()}
                    selectedBlock="some-selected-block"
                    taskStatusEndpoint={taskStatusEndpoint}
                />
            </Provider>,
        );
        const instance = component.root.children[0].instance;
        instance.handleToggleDropdown();
        expect(fetchCourseBlocksMock.mock.calls.length).toBe(1);
    });

    test('display dropdown on toggling dropdown', () => {
        const component = shallow(
            <Main
                courseId={courseId}
                createProblemResponsesReportTask={jest.fn()}
                excludeBlockTypes={excludedBlockTypes}
                fetchCourseBlocks={jest.fn()}
                problemResponsesEndpoint={problemResponsesEndpoint}
                onSelectBlock={jest.fn()}
                selectedBlock="some-selected-block"
                taskStatusEndpoint={taskStatusEndpoint}
            />,
        );
        expect(component.find(BlockBrowserContainer).length).toBeFalsy();
        component.find(Button).find({ label: 'Select a section or problem' }).simulate('click');
        expect(component.find(BlockBrowserContainer).length).toBeTruthy();
    });
});
