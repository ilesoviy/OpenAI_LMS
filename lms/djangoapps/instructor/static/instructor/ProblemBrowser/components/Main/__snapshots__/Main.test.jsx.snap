// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProblemBrowser Main component render with basic parameters 1`] = `
<div
  className="problem-browser-container"
>
  <div
    className="problem-browser"
  >
    <button
      className="btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Select a section or problem
    </button>
    <input
      disabled={true}
      hidden={false}
      name="problem-location"
      type="text"
      value={null}
    />
    <button
      className="btn"
      name="list-problem-responses-csv"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Create a report of problem responses
    </button>
    <div
      aria-live="polite"
      className="report-generation-status"
    />
  </div>
</div>
`;

exports[`ProblemBrowser Main component render with selected block 1`] = `
<div
  className="problem-browser-container"
>
  <div
    className="problem-browser"
  >
    <button
      className="btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Select a section or problem
    </button>
    <input
      disabled={true}
      hidden={false}
      name="problem-location"
      type="text"
      value="some-selected-block"
    />
    <button
      className="btn"
      name="list-problem-responses-csv"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Create a report of problem responses
    </button>
    <div
      aria-live="polite"
      className="report-generation-status"
    />
  </div>
</div>
`;
